lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1035,6 +1035,181 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
+          return failure();
+        if (kernelShape.size()) {
+          if (kernelShape.size() != weightShape.size() - 2) {
+            return rewriter.notifyMatchFailure(
+                binder.op,
+                "unsupported conversion: kernel_shape list size should have "
+                "number of values equal to weight_rank - 2");
+          } else {
+            for (unsigned i = 0; i < kernelShape.size(); i++) {
+              if (weightShape[i + 2] != kernelShape[i]) {
+                return rewriter.notifyMatchFailure(
+                    binder.op, "unsupported conversion: kernel_shape value "
+                               "should be equal to the weight tensor shape");
+              }
+            }
+          }
+        }
+
+        // Determine the rank of input tensor.
+        std::optional<unsigned> maybeRank = Torch::getTensorRank(input);
+        if (!maybeRank)
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Unimplemented: unranked tensor");
+        unsigned rank = *maybeRank;
+
+        SmallVector<int64_t> padding, strides, dilations;
+        SmallVector<int64_t> defaultPadding, defaultStrides, defaultDilations;
+        for (unsigned i = 0; i < rank - 2; i++) {
+          defaultPadding.push_back(0);
+          defaultStrides.push_back(1);
+          defaultDilations.push_back(1);
+        }
+        // Padding for the beginning and ending along each spatial axis, it can
+        // take any value greater than or equal to 0. The value represent the
+        // number of pixels added to the beginning and end part of the
+        // corresponding axis. pads format should be as follow [x1_begin,
+        // x2_begin…x1_end, x2_end,…], where xi_begin the number of pixels added
+        // at the beginning of axis i and xi_end, the number of pixels added at
+        // the end of axis i.
+        if (binder.s64IntegerArrayAttr(padding, "pads", defaultPadding)) {
+          return failure();
+        }
+        if (padding.size() != rank - 2 && padding.size() != 2 * (rank - 2)) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "padding list size does not match the number of axes");
+        }
+        if (binder.s64IntegerArrayAttr(dilations, "dilations",
+                                       defaultDilations)) {
+          return failure();
+        }
+        if (dilations.size() != rank - 2) {
+          return rewriter.notifyMatchFailure(
+              binder.op,
+              "dilations list size does not match the number of axes");
+        }
+        if (binder.s64IntegerArrayAttr(strides, "strides", defaultStrides)) {
+          return failure();
+        }
+        if (strides.size() != rank - 2) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "strides list size does not match the number of axes");
+        }
+
+        Value cstOne;
+        if (binder.op->getNumOperands() > 2) {
```

Comment:
You can remove the `if(binder.op->getNumOperands() > N)` checks Try something like this

```
Value weightZeroPoint;
if (!binder.tensorOperandAtIndex(inputZeroPoint, 2)) {
  cstOne = rewriter.create<Torch::ConstantIntOp>(
    binder.getLoc(), rewriter.getI64IntegerAttr(1));
  auto inputType = input.getType().cast<Torch::ValueTensorType>();
  input = rewriter.create<Torch::AtenSubTensorOp>(
    binder.getLoc(), inputType, input, inputZeroPoint, cstOne);
}
Value weightZeroPoint;
if (!binder.tensorOperandAtIndex(weightZeroPoint, 3)) {
  weight = rewriter.create<Torch::AtenSubTensorOp>(
  binder.getLoc(), weightTensorType, weight, weightZeroPoint,
  cstOne);
}
```